semgrep-core/src/parsing/pfff/scala_to_generic.ml

```diff
@@ -538,10 +538,26 @@ and v_stmt = function
       G.DoWhile (v1, v2, G.unbracket v4) |> G.s
   | For (v1, v2, v3) ->
       let v1 = v_tok v1
-      and _v2TODO = v_bracket v_enumerators v2
+      and v2 = v2 |> G.unbracket |> v_enumerators
       and v3 = v_for_body v3 in
-      let header = G.ForClassic ([], None, None) (* TODO *) in
-      G.For (v1, header, v3) |> G.s
+      List.fold_right
+        (fun gen stmt ->
+          match gen with
+          | `G (pat, tok, e, guards) ->
```

Comment:
Want to leave a comment/helpful doc link explaining Scala for loops for future maintainers?